ground/src/main/java/com/google/android/ground/ui/map/gms/mog/MogClient.kt

```diff
@@ -130,52 +126,57 @@ class MogClient(val collection: MogCollection) {
   }
 
   /** Returns metadata for the MOG bounded by the specified web mercator tile. */
-  private suspend fun getMogMetadata(url: String, bounds: TileCoordinates): MogMetadata? =
-    getMogMetadataAsync(url, bounds).await()
+  private suspend fun getMogMetadata(path: String, bounds: TileCoordinates): MogMetadata? =
+    getMogMetadataAsync(path, bounds).await()
 
   /**
-   * Returns a future containing the [MogMetadata] with the specified extent and URL. Metadata is
+   * Returns a future containing the [MogMetadata] with the specified extent and path. Metadata is
    * either loaded from in-memory cache if present, or asynchronously fetched if necessary. The
    * process of checking the cache and creating the new job is synchronized on the current instance
    * to prevent duplicate parallel requests for the same resource.
    */
-  private fun getMogMetadataAsync(url: String, mogBounds: TileCoordinates): Deferred<MogMetadata?> =
-    synchronized(this) { cache.get(url) ?: getMogMetadataFromRemoteAsync(url, mogBounds) }
+  private fun getMogMetadataAsync(
+    path: String, mogBounds: TileCoordinates
+  ): Deferred<MogMetadata?> =
+    synchronized(this) { cache.get(path) ?: getMogMetadataFromRemoteAsync(path, mogBounds) }
 
   /**
-   * Asynchronously fetches and returns the [MogMetadata] with the specified extent and URL. The
+   * Asynchronously fetches and returns the [MogMetadata] with the specified extent and path. The
    * async job is added to the cache immediately to prevent duplicate fetches from other threads.
    */
   private fun getMogMetadataFromRemoteAsync(
-    url: String,
-    mogBounds: TileCoordinates
+    path: String, mogBounds: TileCoordinates
   ): Deferred<MogMetadata?> = runBlocking {
     // TODO: Exceptions get propagated as cancellation of the coroutine. Handle them!
     async {
-        nullIfNotFound { UrlInputStream(url) }?.use { readMogMetadataAndClose(url, mogBounds, it) }
+      Timber.v("Download path: %s", path)
+      var url: String? = path
```

Comment:
Can we extract this logic into a carefully named fun? For ex "getMogUrlFromPath`? Logging should probably live outside that lower level function.

Reply:
I like the second suggestion below, I prefer util functions over helper methods.

Reply:
Sgtm! Extension functions in Kotlin provide additional syntactic sugar (and potentially improved readability) over traditional static util functions. 🚀